shell/platform/android/io/flutter/embedding/engine/dart/DartMessenger.java

```diff
@@ -33,22 +34,37 @@
 
   @NonNull private final FlutterJNI flutterJNI;
 
-  @NonNull private final ConcurrentHashMap<String, HandlerInfo> messageHandlers;
+  /**
+   * Maps a channel name to an object that contains the task queue and the handler associated with
+   * the channel.
+   *
+   * <p>Reads and writes to this map must lock {@code handlersLock}.
+   */
+  @NonNull private final Map<String, HandlerInfo> messageHandlers = new HashMap<>();
```

Comment:
Instead of two `Map` objects, could we jus tkeep one `ConcurrentHashMap` and either create a wrapper object for the buffered messages or add them to `HandlerInfo`?

Reply:
The wrapper object would still need to represent two different states. One with a Handler, and one without a handler, but with buffered messages.  FWIW, having two separate maps provides a more transparent separation of concerns.